### PR TITLE
Implement Wishlist Functionality

### PR DESCRIPTION
**Pull request by daoanhhuy26012001**
#### Modify the `ProductCard` component:
- [x] Update the `handleHeartClick` event handler to toggle the product's wishlist status using the `addToWishlist` function.
#### Update the `ListProduct` component
- [x] Add the `wishList` and `setWishList` props.
- [x] Modify the `addToWishlist` function to update the wishlist by calling setWishList with the updated list of products.
#### Modify the `WishList` component
- [x] Add the `setWishList` prop.
- [x] Modify the `addToWishlist` function to update the wishlist by calling setWishList with the updated list of products.
#### Update the `App` component
- [x] Add the state variable wishList and setWishList to manage the wishlist.
- [x] Pass the wishList and setWishList props to the ListProduct and WishList components.
- [x] Update the handleWishList function to handle adding and removing products from the wishlist.
#### Test the wishlist functionality:
- [x] Add products to the wishlist by clicking on the heart icon in the `ProductCard` component.
- [x] Verify that the wishlist updates accordingly.
- [x] Toggle the visibility of the wishlist using the appropriate button in the ListCategory component.
- [x] Ensure that the wishlist is displayed correctly with the added products in the WishList component.

### CODE REVIEW

File: ReactJS/practice-one/src/App.jsx [hunk 1]
```diff
@@ -38,32 +42,83 @@ function App() {
...
+        }
+
+        return item;
+      })
+    );
+    if (product.isFavorite) {
+      setWishList((prev) => {
+        return prev.filter((item) => item.id !== product.id);
+      });
+    } else {
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/App.jsx:72]: Dont need else after return

File: ReactJS/practice-one/src/App.jsx [hunk 2]
```diff
@@ -38,32 +42,83 @@ function App() {
...
-    });
+    setFilteredProducts(
+      filteredProducts.filter((product) => {
+        const price = parseInt(product.price.replace(/\./g, ''));
+        return price >= min && price <= max;
+      })
+    );
   }
 
+  // Set filtered products when products change
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/App.jsx:108]: Please add TODO comment, Remove this useEffect

File: ReactJS/practice-one/src/App.jsx [hunk 3]
```diff
@@ -38,32 +42,83 @@ function App() {
...
+  useEffect(() => {
+    if (products.length) {
+      setFilteredProducts(products);
+    }
+  }, [products]);
+
+  // Update wish list when products change
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/App.jsx:115]: Please add TODO comment, Remove this useEffect

File: ReactJS/practice-one/src/components/Common/ProductCard/index.jsx [hunk 4]
```diff
@@ -33,6 +29,14 @@ export const ProductCard = ({
     setSelectedProduct(product);
   };
 
+  /**
+   * Event handler for heart icon click.
+   * Adds the product to the wishlist.
+   */
+  const handleHeartClick = () => {
```
Comment (phattruong-agilityio) [inline ReactJS/practice-one/src/components/Common/ProductCard/index.jsx:36]: Check naming here

Comment (daoanhhuy26012001) [reply ReactJS/practice-one/src/components/Common/ProductCard/index.jsx:36]: Done

**Pull request closed without merging**